configuration/haproxy.go

```diff
@@ -8,4 +10,29 @@ type HAProxy struct {
 	ReloadCleanupCommand    string
 	ShutdownCommand         string
 	GraceSeconds            int
+	HostnameLabel           *string
+}
+
+// BalancerType indicates whether there bamboo is dealing with external or internal traffic
+type BalancerType string
```

Comment:
It would be better to support arbitrary values here-- for instance I'm already considering adding an "oncallers bamboo" to shield the marathon UI behind, and I don't want to have to modify bamboo to do that.